cranelift/filetests/src/lib.rs

```diff
@@ -143,6 +140,15 @@ fn new_subtest(parsed: &TestCommand) -> subtest::SubtestResult<Box<dyn subtest::
         "stack_maps" => test_stack_maps::subtest(parsed),
         "unwind" => test_unwind::subtest(parsed),
         "verifier" => test_verifier::subtest(parsed),
-        _ => Err(format!("unknown test command '{}'", parsed.command)),
+        _ => anyhow::bail!("unknown test command '{}'", parsed.command),
     }
 }
+
+fn pretty_anyhow_error(
```

Comment:
At a slight risk of overengineering, I wonder if we could do something a bit nicer here:

Could we, for example, return a `CodegenErrorWithContext` that wraps the error with the necessary context to interpret it (`func` and `isa`) and return that from the codegen entry points? That would feel a bit better than holding a bare `CodegenError` and recombining it with its context here.

(On second thought, this would force us to plumb lifetime params through in order to carry the func/isa borrows -- maybe too verbose. Maybe there's something else in that spirit though. Needs more thought?)

Reply:
I think ideally we would make `CodegenError` something that can have context applied to it, the same way that the `wast` crate's error type can have source text and filename context applied to it. This would mutate the error, switching it to the string-y form, without holding borrows to the context itself.

https://docs.rs/wast/24.0.0/wast/struct.Error.html#method.set_path
https://docs.rs/wast/24.0.0/wast/struct.Error.html#method.set_text

This is a bit more than I want to tackle right now, so I'll file a follow up issue instead.

Reply:
Filed https://github.com/bytecodealliance/wasmtime/issues/2204